sql/size_utils.sql

```diff
@@ -126,151 +184,280 @@ BEGIN
 
         CASE WHEN is_distributed THEN
-            RETURN QUERY SELECT * FROM _timescaledb_internal.hypertable_relation_remote_size(schema_name, table_name);
+            RETURN QUERY SELECT * FROM _timescaledb_internal.hypertable_remote_size(schema_name, table_name);
         ELSE
-            RETURN QUERY SELECT * FROM _timescaledb_internal.hypertable_relation_local_size(schema_name, table_name);
+            RETURN QUERY SELECT *, NULL::name FROM _timescaledb_internal.hypertable_local_size(schema_name, table_name);
         END CASE;
 END;
 $BODY$;
 
+-- Get relation size of hypertable
+-- like pg_relation_size(hypertable)
+-- (https://www.postgresql.org/docs/9.6/static/functions-admin.html#FUNCTIONS-ADMIN-DBSIZE)
+--
+-- main_table - hypertable to get size of
+--
+-- Returns:
+-- heap_size         - Pretty output of heap_bytes
+-- index_bytes        - Pretty output of index_bytes
+-- toast_bytes        - Pretty output of toast_bytes
+-- total_size         - Pretty output of total_bytes
 
-CREATE OR REPLACE FUNCTION _timescaledb_internal.range_value_to_pretty(
-    time_value      BIGINT,
-    column_type     REGTYPE
+CREATE OR REPLACE FUNCTION hypertable_detailed_size_pretty(
+    main_table              REGCLASS
 )
-    RETURNS TEXT LANGUAGE PLPGSQL STABLE AS
+RETURNS TABLE (heap_size  TEXT,
+               index_size  TEXT,
+               toast_size  TEXT,
+               total_size  TEXT,
+               node_name   NAME) LANGUAGE PLPGSQL STABLE STRICT
+               AS
 $BODY$
 DECLARE
+        table_name       NAME;
+        schema_name      NAME;
 BEGIN
-    IF NOT _timescaledb_internal.dimension_is_finite(time_value) THEN
-        RETURN '';
-    END IF;
-    IF time_value IS NULL THEN
-        RETURN format('%L', NULL);
-    END IF;
-    CASE column_type
-      WHEN 'BIGINT'::regtype, 'INTEGER'::regtype, 'SMALLINT'::regtype THEN
-        RETURN format('%L', time_value); -- scale determined by user.
-      WHEN 'TIMESTAMP'::regtype, 'TIMESTAMPTZ'::regtype THEN
-        -- assume time_value is in microsec
-        RETURN format('%1$L', _timescaledb_internal.to_timestamp(time_value)); -- microseconds
-      WHEN 'DATE'::regtype THEN
-        RETURN format('%L', timezone('UTC',_timescaledb_internal.to_timestamp(time_value))::date);
-      ELSE
-        RETURN time_value;
-    END CASE;
-END
-$BODY$;
+        RETURN QUERY
+        SELECT pg_size_pretty(heap_bytes) as heap_size,
+               pg_size_pretty(index_bytes) as index_size,
+               pg_size_pretty(toast_bytes) as toast_size,
+               pg_size_pretty(total_bytes) as total_size,
+               hd.node_name
+               FROM @extschema@.hypertable_detailed_size(main_table) hd;
 
+END;
+$BODY$;
 
-CREATE OR REPLACE FUNCTION _timescaledb_internal.partitioning_column_to_pretty(
-    d   _timescaledb_catalog.dimension
-)
-    RETURNS TEXT LANGUAGE PLPGSQL STABLE STRICT AS
+CREATE OR REPLACE FUNCTION _timescaledb_internal.chunk_local_size(
```

Comment:
A couple of notes:

* it sounds like this takes a chunk as input, but I think it takes a hypertable. This is confusing.
* why is the input schema+name and not regclass?
* Why does it return chunk_id? I guess this is fine for an internal functions, but not clear why we need it.
* Why do we return only name and not schema (or regclass).

More accurate names would be `chunks_local_size`, `local_chunks_size` or `hypertable_local_chunks_size`

Reply:
Will rename this. Intended to be used by both distributed (remote calls this internally) and local versions. So we pass schema_name + table_name explicitly instead of regclass. 
Also, this relates to efficient access of the catalog tables. We get index access only if we use table_name. Oids are not present in the catalog tables.  Since this is an internal function, I would not worry about this now - can be modified at a later date if use an oid.

Reply:
To be deprecated function chunk_relation_size returns chunk_id. This is useful for  ordering the output. chunk_id gives seq ordering of the chunks. chunk_name has textual ordering and that does not match.